nat/src/main/java/org/hyperledger/besu/nat/docker/HostBasedIpDetector.java

```diff
@@ -13,25 +13,22 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-package org.hyperledger.besu.nat.core;
+package org.hyperledger.besu.nat.docker;
 
-import org.hyperledger.besu.nat.NatMethod;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Optional;
 
-public class AutoDetectionResult {
+public class HostBasedIpDetector implements IpDetector {
 
-  private final NatMethod natMethod;
-  private final boolean isDetectedNatMethod;
+  private static final String HOSTNAME = "HOST_IP";
 
-  public AutoDetectionResult(final NatMethod natMethod, final boolean isDetectedNatMethod) {
-    this.natMethod = natMethod;
-    this.isDetectedNatMethod = isDetectedNatMethod;
-  }
-
-  public NatMethod getNatMethod() {
-    return natMethod;
-  }
-
-  public boolean isDetectedNatMethod() {
-    return isDetectedNatMethod;
+  @Override
+  public Optional<String> detectExternalIp() {
+    try {
+      return Optional.of(InetAddress.getByName(HOSTNAME).getHostAddress());
```

Comment:
How does this behave if you don't supply the `HOST_IP` env var to the docker command?

Reply:
if you don't supply the `HOST_IP` en var the NatManager will try to retrieve the value of the `p2p-host` CLI parameter given when besu started. If this parameter is not given it will be the default value of `p2p-host` as described in the code below.

``` 
  private String p2pHost = autoDiscoverDefaultIP().getHostAddress();
```

To sum up: 
   1)  `HOST_IP` env var
   2) `p2p-host` CLI parameter given when besu started
   3) default value of `p2p-host`